workspace/core/server/src/datastores/content/comments.rs

```diff
@@ -0,0 +1,296 @@
+            .prepare_cached("select * from metadata_comments where id = $1")
+            .await?;
+        let rows = connection
+            .query(&stmt, &[id])
+            .await?;
+        Ok(rows.first().map(|r| r.into()))
+    }
+
+    #[tracing::instrument(skip(self, profile_id, metadata_id, version, offset, limit))]
+    pub async fn get_metadata_comments(
+        &self,
+        profile_id: &Option<Uuid>,
+        metadata_id: &Uuid,
+        version: &i32,
+        manager: bool,
+        offset: i64,
+        limit: i64,
+    ) -> Result<Vec<Comment>, Error> {
+        let connection = self.pool.get().await?;
+        let rows = if manager {
+            let stmt = connection
+                .prepare_cached("select * from metadata_comments where metadata_id = $1 and version = $2 and deleted = false order by created desc offset $3 limit $4")
+                .await?;
+            connection
+                .query(&stmt, &[metadata_id, &version, &offset, &limit])
+                .await?
+        } else if let Some(profile_id) = profile_id {
+            let stmt = connection
+                .prepare_cached("select * from metadata_comments where metadata_id = $1 and version = $2 and deleted = false and ((visibility = 'public' and status = 'approved') or (profile_id = $3)) order by created desc offset $4 limit $5")
+                .await?;
+            connection
+                .query(&stmt, &[metadata_id, &version, profile_id, &offset, &limit])
+                .await?
+        } else {
+            let stmt = connection
+                .prepare_cached("select * from metadata_comments where metadata_id = $1 and version = $2 and deleted = false and (visibility = 'public' and status = 'approved') order by created desc offset $3 limit $4")
+                .await?;
+            connection
+                .query(&stmt, &[metadata_id, &version, &offset, &limit])
+                .await?
+        };
+        Ok(rows.iter().map(|r| r.into()).collect())
+    }
+
+    #[tracing::instrument(skip(self, profile_id, metadata_id, version))]
+    pub async fn get_metadata_comments_count(
+        &self,
+        profile_id: &Option<Uuid>,
+        metadata_id: &Uuid,
+        version: &i32,
+        manager: bool,
+    ) -> Result<i64, Error> {
+        let connection = self.pool.get().await?;
+        let row = if manager {
+            let stmt = connection
+                .prepare_cached("select count(*) from metadata_comments where metadata_id = $1 and version = $2 and deleted = false")
+                .await?;
+            connection
+                .query_one(&stmt, &[metadata_id, &version])
+                .await?
+        } else if let Some(profile_id) = profile_id {
+            let stmt = connection
+                .prepare_cached("select count(*) from metadata_comments where metadata_id = $1 and version = $2 and deleted = false and ((visibility = 'public' and status = 'approved') or (profile_id = $3)) desc")
```

Comment:
```suggestion
                .prepare_cached("select count(*) from metadata_comments where metadata_id = $1 and version = $2 and deleted = false and ((visibility = 'public' and status = 'approved') or (profile_id = $3)) order by created desc")
```

Looks to be an orphaned desc. Suggest adding order by created desc